x/gov/types/proposal.go

```diff
@@ -359,3 +238,121 @@ func (tr TallyResult) String() string {
   No:         %s
   NoWithVeto: %s`, tr.Yes, tr.Abstain, tr.No, tr.NoWithVeto)
 }
+
+// Proposal types
+const (
+	ProposalTypeText            string = "Text"
+	ProposalTypeSoftwareUpgrade string = "SoftwareUpgrade"
+)
+
+// Text Proposal
+type TextProposal struct {
+	Title       string `json:"title"`
+	Description string `json:"description"`
+}
+
+func NewTextProposal(title, description string) Content {
+	return TextProposal{title, description}
+}
+
+// Implements Proposal Interface
+var _ Content = TextProposal{}
+
+// nolint
+func (tp TextProposal) GetTitle() string         { return tp.Title }
+func (tp TextProposal) GetDescription() string   { return tp.Description }
+func (tp TextProposal) ProposalRoute() string    { return RouterKey }
+func (tp TextProposal) ProposalType() string     { return ProposalTypeText }
+func (tp TextProposal) ValidateBasic() sdk.Error { return ValidateAbstract(DefaultCodespace, tp) }
+
+func (tp TextProposal) String() string {
+	return fmt.Sprintf(`Text Proposal:
+  Title:       %s
+  Description: %s
+`, tp.Title, tp.Description)
+}
+
+// Software Upgrade Proposals
```

Comment:
Aren't these disabled (mentioned in the docs) - are we just leaving the code in?

Reply:
They are indeed disabled and yes I just left the existing code intact. Do you recommend we remove it?

Reply:
It's ok, just checking